### DISCUSSION TITLE
Should the `reflectance` field of `StandardMaterial` be renamed?

### DISCUSSION THREAD

**Discussion by nicopap** [bevyengine/bevy#5936] [General]
As the [filament] doc points out, "reflectance" as a shader parameter is not identical to the physical quantity "reflectance" (a value of `0.5` resulting in a `4%` reflectance, ie `0.04`) In any case, that value is mostly used to control the specular intensity of the material.

"Reflectance" is a _very_ niche term that most users are not familiar with, while "specular intensity," is explicit to anyone who knows what "specular" means. Question becomes: should we rename `reflectance` to `specular_intensity`?

[filament]: https://google.github.io/filament/Material%20Properties.pdf

**Comment by superdump** [comment]
I’d argue that in English, reflect is a vastly more common word than specular. I’m not sure I’ve heard the word specular outside of rendering/computer graphics. And reflectance is a conjugation (form of a verb) of reflect. Reflectance sounds like how reflective something is. BRDF is bidirectional reflectance distribution function. So in summary I don’t think it should be changed from reflectance to specular intensity.

That said, we should perhaps clarify how it is different from physical quantities both through its name and documentation. And we should document how to think about it from an artistic perspective. The filament documentation of how to think about the materials with visual examples is pretty useful.

And all of that said, I am interested to learn what content creation tools output that would map to reflectance. I was expecting to see it as part of the gltf metallic roughness material model but I don’t see it in our gltf loader at least. Hmm.

**Reply by superdump** [reply]
I wasn't keen on calling it specular intensity, but now I've seen the Filament material properties description referring to it as such, I'm changing my mind. However, elsewhere @djeedai suggested that 'specular intensity' may be a term for a coefficient used in phong shading. I'm not sure if it maps exactly to that term or not. If it does then it feels like it may make sense to rename it and explain the relationship between the two.

**Reply by djeedai** [reply]
My concern is not on the terminology, where the rendering team has vastly more experience and can inform a decision. The issue I pointed is the contradiction of having a field named `reflectance` whose documentation defines it as "mapping to the reflectance". In short, the docs read as "field X is a Y that internally maps to X", which is quite confusing. I have no opinion on the term itself we choose, I just want the documentation to avoid that contradiction.

**Reply by djeedai** [reply]
...and since I understand that the docs may be correct in explaining what Bevy actually does, my logical conclusion was that the name was wrong (should have been Y).

**Reply by aloucks** [reply]
> I’m not sure I’ve heard the word specular outside of rendering/computer graphics.

This is true, but it can be confusing when the majority of other books and articles use that vocabulary. Does the bevy documentation indicate that the `reflectance` property is often called `specular intensity`?

**Comment by nicopap** [comment]
Other considerations: people may be able to infer what "reflectance" means based on similarity with "reflection", but are not necessarily familiar with the actual physical measure "reflectance". Meaning that a user may expect it to do what it does, without having expectations about the effect of a particular value has.
